src/components/side_panel/components/collapse/collapse.ts

```diff
@@ -0,0 +1,50 @@
+import { Component, onMounted, onWillUpdateProps, useRef } from "@odoo/owl";
+import { SpreadsheetChildEnv } from "../../../../types";
+
+interface Props {
+  isCollapsed: boolean;
+  slots: any;
+}
+
+export class Collapse extends Component<Props, SpreadsheetChildEnv> {
+  static template = "o-spreadsheet-Collapse";
+  static props = {
+    isCollapsed: Boolean,
+    slots: Object,
+  };
+
+  private contentRef = useRef("content");
+
+  setup() {
+    onMounted(() => {
+      if (this.props.isCollapsed) {
+        this.contentRef.el?.classList.add("d-none");
+      }
+    });
+    onWillUpdateProps((nextProps) => {
```

Comment:
I'm not sure `onWillUpdateProps` is appropriate because it starts the transition *before* the component is rendered. The element could technically be changed (even it probably never happens in practice here)